### PR TITLE
Fixing camera search bar

### PR DESCRIPTION
**Pull request by caesarx26**
# Pull Request Template

## Description
I fixed bugs with the search bar to have it show correctly and be hidden when categories are selected. 

I also fixed the camera modal to show up when a button is pressed in the footer. And in the modal a function is passed into it to set the state to close the modal when the cancel button is pressed or the user clicks anywhere outside the modal. And changed the styling.

The modal buttons to take a picture and get a picture are disabled when first pressed and then enabled when they are done doing the request or if the person cancels the request to take a picture or get a picture. This is to prevent the user clicking a button more than one at a time so no multiple pops up to get a picture occur. 

I also fixed issues with the linting need to have jest in the environments and also needed to install the react testing library so I installed that and set jest to be used in the environment for eslint

## Type of Change

- [x] Bugfix
- [ ] New Feature
- [ ] Documentation Update
- [ ] Other (please describe):

## Checklist

- [x] I have read the contributing guidelines.
- [ ] I have followed the code style of this project.
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation (if applicable).
- [x] My changes do not generate new warnings.

## Additional Notes
- the search bar shows up and mostly works now but it will sometimes show additional items than the ones searched for on all pages like typing "b" pulls up 1 book even tho none of the books have the letter b in it.
- Will need to test filtering of the search bar more to see if it is accurate

Addresses Issue #29 

## images
Camera modal
categories
categories pinned and selected

### CODE REVIEW

**Comment by MrSnoodle01** [comment]
Overall looks good and am happy that the lint issues are fixed. 

But one thing I am curious about is why we got rid of the blur behind the camera. I thought that the blur looked nice and put the focus onto the camera pop up. 

Also it would be nice if you could make an issue for the search bar not working properly.

File: Tabby/components/FooterNavBar.tsx [hunk 1]
```diff
@@ -1,61 +1,77 @@
...
 import SelectedFavorite from "@/assets/navbar-images/selectedFavorite.svg";
 import NotSelectedFavorite from "@/assets/navbar-images/notSelectedFavorite.svg";
 import Settings from "@/assets/navbar-images/settings.svg";
+import CameraIcon from "@/assets/navbar-images/camera.svg";
+import CameraModal from "@/components/camera/CameraModel";
 
 const FooterNavBar = () => {
-  // getting the current page to see the path and using that to make the icons change to selected if that path invloves the page that icon is related to 
+  // set to true to show camera modal
+  const [isCameraModalVisible, setCameraModalVisible] = useState(false);
```
Comment (LilAttacker) [inline Tabby/components/FooterNavBar.tsx:16]: > set to true to show camera modal

I feel like this code is doing the opposite of what the comment says. Can this be checked?

File: Tabby/app/recommendations/index.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ const initialBooks: Book[] = [
     },
     {
         id: '2',
-        title: 'To Kill a Mockingbird',
+        title: 'To Kill',
```
Comment (LilAttacker) [inline Tabby/app/recommendations/index.tsx:30]: I get what you're doing here, changing the names so the search bar can be more properly tested.

But I feel keeping the names this similar may be hurting that. I think the changes should be more substantial and you can just different book names.

Some examples could be:
- Lord of the Rings
- Game of Thrones
- Twilight
- The Bible

Just take a look at these series: https://www.forbes.com/sites/entertainment/article/best-book-series/
If  you want more exact book names: https://www.barnesandnoble.com/b/books/_/N-1fZ29Z8q8

The only thing that really needs to change is the title. The description and author can stay the same unless you plan to add a filter for the author or make it to where you can search by author.

File: Tabby/app/recommendations/[book].tsx [hunk 3]
```diff
@@ -87,13 +87,13 @@ const BookPage = () => {
...
-        console.log(`Deleting ${book} from ${category}`);
+        console.log(`Deleting ${book} from reccomendations`);
     };
 
     // Extract both category and book slugs
-    const { category, book } = useLocalSearchParams();
+    const { book } = useLocalSearchParams();
 
-    console.log(book, "is from", category);
+    console.log(book, "is reccomended");
```
Comment (LilAttacker) [inline Tabby/app/recommendations/[book].tsx:96]: > reccomended

This is a type. It is typed recommend.

Comment (LilAttacker) [reply Tabby/app/recommendations/[book].tsx:96]: This also seems to be an issue in many places.

**Review by LilAttacker — CHANGES_REQUESTED** [review]
Excellent that this fixes the jest issue. I hope a few things can be addressed before merging though.

First, I'm in agreement with Josh. Not sure what happened to the blur when you tapped the camera button, but I think that was a good feature and should be kept.

Second, please explain why most of the comments for Tabby/app/library/index.tsx were removed and not replaced. I don't feel that was a necessary change, but if there's an explanation then I'll hear it.

Third, please check my other comment reviews. I don't care much if they aren't implemented, but please at least address them.

--

I will also say, I think both Tabby/components/FooterNavBar.tsx and Tabby/components/camera/CameraModel.tsx could use some more documentation. Whether added or not won't change my review, but I think it'll be a good thing for the project if it was included.

**Review by David1000734 — APPROVED** [review]
Glad we were able to get the lint issue fixed. The comment out code in the test can be removed to clean up the file a little more.

**Review by MrSnoodle01 — APPROVED** [review]

**Pull request merged**